### PR TITLE
feat: add ability to run tests

### PR DESCRIPTION
**Pull request by DudaGod**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,61 @@
...
+
+![Testing view](./images/testing-view.png "Testing view")
+
+You can access the extension from the Testing view in the Visual Studio Code sidebar.
+
+The toolbar at the top provides various commands to manage test execution:
+- **Run Failed Tests**: to run only failed tests (displayed only if there are failed tests in the run);
+- **Refresh Tests**: to reload your tests list, reflecting any new changes;
+- **Run All Tests**: to run all tests that are currently visible;
+- **Debug Tests**: is not supported yet;
```
Comment (sipayRT) [inline README.md]: so its not provided and should be removed

File: src/api/runner.ts [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    TestplaneMasterRunTestsOpts,
+} from "../types";
+
+interface LocationFromStack {
+    path: string;
+    line: number;
+    column: number;
+}
+
+enum TestStatuses {
```
Comment (sipayRT) [inline src/api/runner.ts:26]: what about ERROR status?

File: src/commands/installer.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    const terminal = vscode.window.createTerminal({
+        name: "Install Testplane",
+        cwd: workspaceFolder.uri.fsPath,
+        env: process.env,
+    });
+
+    terminal.show();
+
+    terminal.sendText(`npm init testplane@latest -- --yes`, true);
```
Comment (sipayRT) [inline src/commands/installer.ts]: option `--yes` is `true` by default now

File: src/test-tree/data/storage.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import type * as vscode from "vscode";
+import type { TestData } from "./types";
+
+export const getTestData = (item: vscode.TestItem): TestData => {
+    const data = TEST_DATA_BY_ITEM.get(item);
+
+    if (!data) {
+        throw new Error(
+            `Test data not found for "${item.label}". Please report it to https://github.com/gemini-testing/testplane`,
```
Comment (sipayRT) [inline src/test-tree/data/storage.ts]: https://github.com/gemini-testing/testplane/issues

File: src/utils/error.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import * as vscode from "vscode";
+import { logger } from "./logger";
+
+export const showTestplaneError = (message: string, error?: Error): void => {
+    if (error) {
+        logger.error(error);
+    }
+
+    vscode.window.showErrorMessage(`${message}. Check the output for more details.`, "See error").then(result => {
+        if (result === "See error") {
```
Comment (sipayRT) [inline src/utils/error.ts]: but you added this error text by yourself. And then you'll look at it

File: src/api/runner.ts [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        private readonly _tree: TestTree,
+        private readonly _workspaceFolder: vscode.WorkspaceFolder,
+        private readonly _api: TestplaneMasterRPC,
+        private readonly _handlers: TestplaneMasterHandlers,
+    ) {
+        super(() => {
+            logger.debug?.("Dispose test runner");
+
+            this._handlers.clearListeners();
+            this._endTestRun();
```
Comment (sipayRT) [inline src/api/runner.ts:48]: are we trying to end runner to be sure that there are definitely no other launches?

File: src/worker/index.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import type Testplane from "testplane";
+
+type MessageWithError = {
+    e: Error;
+};
+
+ipc.on<MasterInitEventMessage>(MasterEvents.INIT, message => {
+    try {
+        const { cwd, configPath } = message;
+        const testplaneNodePath = require.resolve("testplane", { paths: [cwd] });
```
Comment (sipayRT) [inline src/worker/index.ts]: do you remember about yarn and its custom paths? :)

**Review by sipayRT — APPROVED** [review]

File: src/api/runner.ts (hunk 2 above)
Comment (DudaGod) [reply src/api/runner.ts:26]: Here I describe only the test completion events. Moreove for any error in the test, the `TEST_FAIL` event is emitted - https://github.com/gemini-testing/testplane/blob/master/src/runner/test-runner/regular-test-runner.js#L39-L44

File: src/api/runner.ts (hunk 6 above)
Comment (DudaGod) [reply src/api/runner.ts:48]: Yes. In this method I should run commands to clear the state

File: src/worker/index.ts (hunk 7 above)
Comment (DudaGod) [reply src/worker/index.ts]: I forgot to fix it here, thanks

**Pull request merged**